### PR TITLE
fix: always add commit to dataset if possible

### PR DESCRIPTION
**Pull request by rokroskar**
Always add a `commit` to `Dataset` if possible - in the case where a dataset is first being created with `with_dataset`, it won't have been committed yet.

This addresses #646 but needs the resolution of #639 for a full solution. 

- [x] I have run ``./run-tests.sh`` locally.

### CODE REVIEW

File: renku/models/_jsonld.py [hunk 1]
```diff
@@ -313,9 +313,9 @@ class JSONLDMixin(ReferenceMixin):
     def from_jsonld(
         cls,
         data,
-        client=None,
         __reference__=None,
         __source__=None,
+        **kwargs,
```
Comment (jsam) [inline renku/models/_jsonld.py]:
```suggestion
        **kwargs
```

File: renku/models/_jsonld.py [hunk 2]
```diff
@@ -372,17 +370,17 @@ def from_jsonld(
         return self
 
     @classmethod
-    def from_yaml(cls, path, client=None):
+    def from_yaml(cls, path, **kwargs):
```
Comment (jsam) [inline renku/models/_jsonld.py]: I generally think kwargs should only be used as a last resort. Can we be explicit with our params or at least document what can come in kwargs?

Comment (rokroskar) [reply renku/models/_jsonld.py]: Sure, it was just starting to look wordy but you're probably right

File: renku/models/_jsonld.py [hunk 3]
```diff
@@ -331,7 +332,7 @@ def from_jsonld(
             ) != type_:
                 new_cls = cls.__type_registry__[type_]
                 if cls != new_cls:
-                    return new_cls.from_jsonld(data, client=client)
+                    return new_cls.from_jsonld(data, client=client, commit=commit)
```
Comment (jsam) [inline renku/models/_jsonld.py]:
```suggestion
                    return new_cls.from_jsonld(
                        data, 
                        client=client, 
                        commit=commit
                    )
```

**Pull request merged**